.gitignore

```diff
@@ -0,0 +1,162 @@
+# Byte-compiled / optimized / DLL files
```

Comment:
In general I prefer each person put things like this in a global `.gitignore` file, rather than copy this into every repo:

```bash
f=$HOME/.gitignore.global  # name this whatever you want
git config --global core.excludesfile $f
cat >$f <<EOF
# patterns to .gitignore by default, across all repos
EOF
```

This is fine, just noting.

Reply:
Understood. I was attempting to follow the `TileDB-SOMA` repo conventions, which seems to have the config completely in the repo.  We can certainly thin out this file in another PR if you think that would be of benefit.